File: chapter02/udp_client/udp_client.py
import socket

host = "127.0.0.1"
port = 9999

client = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

client.sendto(b"AAABBBCCC", (host,port))

data, addr = client.recvfrom(4096)

print(data.decode())
print(addr)

client.close()

File: chapter02/udp_client/udp_client.go
package client

import (
	"fmt"
	"net"
)

func UDPClient() {
	targetHost := "192.168.31.44"
	targetPort := ":9999"

	conn, err := net.Dial("udp", targetHost+targetPort)
	if err != nil {
		fmt.Println(err)
		return
	}

	defer conn.Close()

	_, err = conn.Write([]byte("AAABBBCCC"))
	if err != nil {
		fmt.Println(err)
		return
	}

	var data [4096]byte
	_, err = conn.Read(data[:])
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println(string(data[:]))
	fmt.Println(conn.RemoteAddr())
}
